impala/datadog_checks/impala/check.py

```diff
@@ -0,0 +1,78 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import re
+
+from datadog_checks.base import ConfigurationError, OpenMetricsBaseCheckV2
+from datadog_checks.impala.config_models import ConfigMixin
+from datadog_checks.impala.metrics_catalog import CATALOG_METRIC_MAP, CATALOG_METRICS_WITH_LABEL_IN_NAME
+from datadog_checks.impala.metrics_daemon import DAEMON_METRIC_MAP, DAEMON_METRICS_WITH_LABEL_IN_NAME
+from datadog_checks.impala.metrics_statestore import STATESTORE_METRIC_MAP
+
+TO_SNAKE_CASE_PATTERN = re.compile('(?!^)([A-Z]+)')
+
+
+class ImpalaCheck(OpenMetricsBaseCheckV2, ConfigMixin):
+    __NAMESPACE__ = 'impala'
+
+    def __init__(self, name, init_config, instances):
+        super().__init__(name, init_config, instances)
+        self.check_initializations.append(self.configure_additional_transformers)
+
+    def get_default_config(self):
+        if self.instance["service_type"] == "daemon":
+            return {
+                "metrics": [DAEMON_METRIC_MAP],
+            }
+
+        if self.instance["service_type"] == "statestore":
+            return {
+                "metrics": [STATESTORE_METRIC_MAP],
+            }
+
+        if self.instance["service_type"] == "catalog":
+            return {
+                "metrics": [CATALOG_METRIC_MAP],
+            }
+
+        # Should not happen because this is validated at boot
+        raise ConfigurationError(
+            f"service_type unexpected value ({self.instance['service_type']}); "
+            f"permitted: 'daemon', 'statestore', 'catalog'"
+        )
+
```

Comment:
Could you find a metric that represents the impala version so that we can submit that metadata?

Reply:
Nope, we do not have this info from the openmetrics endpoints unfortunately. This is only available on the debug UI.